### DISCUSSION TITLE
[Feature] Merge Faces sorted by Similarity

### DISCUSSION THREAD

**Discussion by cfelicio** [immich-app/immich#4347] [Feature Request]
### The feature

When attempting to merge faces, it would be useful that instead of displaying the available options in the default order, to see them in a descending order of similarity (so faces that are a close match to the algorithm but not close enough to be grouped together would show at the top). 

For someone trying to sort thru hundreds of faces, this can turn an impossible, arduous task, into a more manageable one. Thoughts?

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by akropp** [comment]
Yes, this would be awesome!  Seems like a straightforward change.  Possibly as another button next to "Show & hide faces".

**Reply by chodthewacko** [reply]
This would be huge.     Even for unassigned faces, It would be a huge QoL improvement to have an box in 'info'  where it showed you the named faces that are the closest match to this face.   This doesn't have to be real time, it could be regenerated via an option in facial recognition.

**Comment by stanstrup** [comment]
This really would be a huge improvement. I started using immich on 60.000 pictures. I have many, many duplicates because I had to be conservative since all the children in my family looks very similar... Having the similar faces close would make it a lot easier to sort out.

**Comment by ASTRELION** [comment]
I'd also like to add that it would be nice for this feature to be some form of wizard that I can go through for the most similar faces, like how Google Photos does it. It makes it much easier if I can just be prompted to go through people that might have duplicates rather than manually matching faces or even having to go through the people menu at all. (I opened https://github.com/immich-app/immich/discussions/14420 but it was essentially this feature request)

**Comment by cfelicio** [comment]
For those watching this thread, looks like we got a nice gift before the Immich team wraps up for the year! The feature was added to their final 2024 release, v1.123.0. You can check the details here: https://github.com/immich-app/immich/pull/14635
